tests/openai/mock_openai.py

```diff
@@ -79,6 +80,11 @@ async def chat_response_stream():
 
 @app.post("/chat/completions")
 async def chat(payload: ChatPayload):
+    if not 0.0 <= payload.temperature <= 2.0:
+        return fastapi.Response(
+            content="Temperature must be between 0.0 and 2.0",
+            status_code=400,
+        )
```

Comment:
this lets us test error behaviours. `openai` will throw a `openai.BadRequestError` when a 400 response is returned